vignettes/ChromBand.Rmd

```diff
@@ -0,0 +1,437 @@
+
+Figure [1](#fig:chr12ideogram){reference-type="ref" reference="fig:chr12ideogram"} shows an ideogram of human chromosome 12, with the band 12q21 shaded. As shown in the figure, 12q21 can be divided into more granular levels 12q21.1, 12q21.2, and 12q21.3. 12q21.3 can itself be divided at an even finer level of resolution into 12q21.31, 12q21.32, and 12q21.33. Moving towards less granular bands, 12q21 is a part of 12q2 which is again a part of 12q. We take advantage of this nested structure of the bands in our analysis.
+
+
+```{r Category, message=FALSE}
+
+library("Category") 
+library("ALL") 
+library("hgu95av2.db")
+library("annotate") 
+library("genefilter") 
+##library("SNPchip")
+library("karyoploteR") 
+library("geneplotter") 
+library("limma") 
+library("lattice")
+library("graph")
+
+```
+
+<figure class="center" id="fig:chr12ideogram">
+```{r chr12ideogram, fig=TRUE, fig.align="center", echo=FALSE, prefix=FALSE, width=9, height=3}
+
+build <- "hg19" ## or "hg18"
+kp <- plotKaryotype(genome=build, chromosomes="chr12", main="Human chromosome12" )
+## cyt2 <- getCytoband(build=build)
+## cyt2$gieStain <- "foo"
+## c12p12_idx <- intersect(grep("^q21", cyt2$name),
+##                         which(cyt2$chrom == "12"))
+## cyt2[c12p12_idx, "gieStain"] <- rep(c("gpos50", "gpos75"),
+##                                     length=length(c12p12_idx))
+##
+##
+## plotCytoband2(chromosome="12", build=build, cytoband=cyt2, outer=FALSE,  ## From SNPchip
+##     cex.axis=0.6, main="Human chromosome 12")
+## plotCytoband2(chromosome="12", build=build, cytoband=cyt2, outer=FALSE,
+##               cex.axis=0.6,
+##               main="Human chromosome 12")
+
+```
+<figcaption>Ideogram for human chromosome 12. The p arm is on the left, the q arm is on the right, and the centromere is indicated by a notch. The shaded bands together represent 12q21. This band is composed of three sub-bands: 12q21.1, 12q21.2, and 12q21.3. The last of these is composed of sub-sub-bands 12q21.31, 12q21.32, and 12q21.33.</figcaption>
+</figure>
```

Comment:
This is the code you shared with me on slack:

    <a id="fig:chr12ideogram">
    ```{r chr12ideogram, echo=FALSE, prefix=FALSE, fig=TRUE, fig.align="center", fig.width=9, fig.height=3, fig.cap="Ideogram for human chromosome 12."}
    build <- "hg19" ## or "hg18"
    kp <- plotKaryotype(genome=build, chromosomes="chr12", main="Human chromosome12" )
    ## cyt2 <- getCytoband(build=build)
    ## cyt2$gieStain <- "foo"
    ## c12p12_idx <- intersect(grep("^q21", cyt2$name),
    ##                         which(cyt2$chrom == "12"))
    ## cyt2[c12p12_idx, "gieStain"] <- rep(c("gpos50", "gpos75"),
    ##                                     length=length(c12p12_idx))
    ##  
    ##  
    ## plotCytoband2(chromosome="12", build=build, cytoband=cyt2, outer=FALSE,  ## From SNPchip
    ##     cex.axis=0.6, main="Human chromosome 12")
    ## plotCytoband2(chromosome="12", build=build, cytoband=cyt2, outer=FALSE,
    ##               cex.axis=0.6,
    ##               main="Human chromosome 12")
    ``` 
    The p arm is on the left, the q arm is on the right, and the centromere is indicated by a notch. The shaded bands together represent 12q21. This band is composed of three sub-bands: 12q21.1, 12q21.2, and    +++12q21.3. The last of these is composed of sub-sub-bands 12q21.31, 12q21.32, and 12q21.33.
    </a>

Reply:
You need to include all of the caption. So the part that starts `The p arm is on the left` should also be inside `fig.cap`.

I also asked you to remove `prefix=FALSE, fig=TRUE, `; I know they came from the .Rnw but I don't know if they are still used or what they do. I don't see them in available options for knitr or R Markdown.

The anchor works. When you click on `1` in the above paragraphs.

Reply:
`<a id="fig:chr12ideogram">
```{r chr12ideogram, echo=FALSE, prefix=FALSE, fig=TRUE, fig.align="center", fig.width=9, fig.height=3, **fig.cap="Ideogram for human chromosome 12.The p arm is on the left, the q arm is on the right, and the centromere is indicated by a notch. The shaded bands together represent 12q21. This band is composed of three sub-bands: 12q21.1, 12q21.2, and 12q21.3. The last of these is composed of sub-sub-bands 12q21.31, 12q21.32, and 12q21.33."}**

build <- "hg19" ## or "hg18"
kp <- plotKaryotype(genome=build, chromosomes="chr12", main="Human chromosome12" )
## cyt2 <- getCytoband(build=build)
## cyt2$gieStain <- "foo"
## c12p12_idx <- intersect(grep("^q21", cyt2$name),
##                         which(cyt2$chrom == "12"))
## cyt2[c12p12_idx, "gieStain"] <- rep(c("gpos50", "gpos75"),
##                                     length=length(c12p12_idx))
##
##
## plotCytoband2(chromosome="12", build=build, 
cytoband=cyt2, outer=FALSE,  ## From SNPchip
##     cex.axis=0.6, main="Human chromosome 12")
## plotCytoband2(chromosome="12", build=build, cytoband=cyt2, outer=FALSE,
##               cex.axis=0.6,
##               main="Human chromosome 12")
```
</a>`
As per your instructions, I did that and it  produce this:
![Screen Shot 2023-03-30 at 8 12 42 PM](https://user-images.githubusercontent.com/57273086/228927510-3349f70a-d879-4210-9b6d-fab62382a172.png)

Reply:
> If the anchor underlines, I suggest just doing `<a id="fig:chr12ideogram"></a>` above the figure. Probably the `a` tags are styled to look like links. Doing as I suggest will still work because we just need it to jump to the image.

Ok, just to clarify, I must do this: <a id="fig:chr12ideogram"> <figure>...</figure></a>

Reply:
Please make all requested changes. If you are still having an issue with section, share the code so that I can replicate the problem.